src/components/Editor/AddFileModal.tsx

```diff
@@ -0,0 +1,119 @@
+import { Dialog } from '@headlessui/react';
+import { useSetAtom } from 'jotai';
+import React, { useContext, useRef, useState } from 'react';
+import Select from 'react-select';
+import { createNewInternalSolutionFileAtom } from '../../atoms/editor';
+import { DarkModeContext } from '../../context/DarkModeContext';
+import { AlgoliaEditorSolutionFile } from '../../models/algoliaEditorFile';
+import Modal from '../Modal';
+import parse from './parsers/parse';
+export default function AddFileModal(props) {
+  const darkMode = useContext(DarkModeContext);
+  const [division, setDivision] = useState('');
+  const [fileStatus, setFileStatus] = useState('Create File');
+  const fileURLRef = useRef<HTMLInputElement>(null);
+  const createSol = useSetAtom(createNewInternalSolutionFileAtom);
+  console.log(division);
+  return (
+    <Modal {...props}>
+      <Dialog.Panel className="bg-white dark:bg-black w-full max-w-xl dark:text-white p-5 rounded-lg shadow-lg flex flex-col items-start">
+        <h3 className="text-lg font-bold">Enter Problem URL</h3>
+        <input
+          type="url"
+          className="shadow-sm focus:ring-blue-500 focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md dark:bg-gray-900 dark:border-gray-700"
+          placeholder="e.g. https://codeforces.com/contest/1920/problem/C"
+          ref={fileURLRef}
+        />
+        <p className="mt-2">Problem Division</p>
+        <div className="mt-2 relative w-full dark:bg-black rounded-md shadow-sm">
+          <Select
+            options={[
+              'General',
+              'Bronze',
+              'Silver',
+              'Gold',
+              'Platinum',
+              'Advanced',
+            ].map(div => ({
+              label: div,
+              value: div.toLowerCase(),
+            }))}
+            onChange={e => setDivision(e.value)}
+            className="tw-forms-disable text-left"
+            styles={
+              !darkMode
+                ? undefined
+                : {
+                    control: provided => ({
+                      ...provided,
+                      backgroundColor: '#111827',
+                      borderColor: '#374151',
+                    }),
+                    menuList: provided => ({
+                      ...provided,
+                      borderColor: '#374151',
+                      borderWidth: '1px',
+                      borderRadius: '6px',
```

Comment:
I feel like this code is duplicated in multiple places in the codebase (https://github.com/cpinitiative/usaco-guide/blob/bbc8e5b44a4eb3f6f080672b02f76a950801045a/src/components/ProblemSuggestionModal.tsx#L250), (https://github.com/cpinitiative/usaco-guide/blob/bbc8e5b44a4eb3f6f080672b02f76a950801045a/src/components/ProblemsPage/Selection.tsx#L73), maybe more. we may want to clean it up sometime

Reply:
it's not immediately related to this PR though, so feel free to do this in another PR if you don't want to do it now

Reply:
refactored!